### ISSUE TITLE
Next steps to make this code for realtime scenarios

### ISSUE THREAD

**Issue by pavankumarkatakam** [davidsandberg/facenet#313]
Hi @davidsandberg, First of all, thanks for your efforts putting the paper into tensorflow code and it is working well.

 This is not a bug rather a discussion on next steps to make this software ready for real time scenarios lets say Office Attendance system. 

I have studied the WIKI and trained the model on LFW dataset and tested also getting the 1.000 accuracy. I have also tested on my own dataset of images(Tollywood film actors) which is successful. 
So my next step to train the SVM which is already trained I mean first i have trained the 20 actor faces but now I want to add 5 more actresses how do i do this rather doing the whole training again on 15 members.

From the prediction part, I can understand that we need to keep a threshold to get the unknown face results.

**Comment by vudung45** [comment]
There are two approaches for this:
You can either get the embeddings ( 128D vectors) and turn them into input for a new simple neural network to classify the 20 "actors' faces" ( simple 2 layers neural network)

Another simple but very effective approach is to store all the embeddings for each of your actor, then do a simple linear min Ellucian distance search on all pre-existing data. The one with the min distance is most likely to be the person. For unknown face results, you can keep a threshold for the distance, if the found min distance is larger than that then the person is unknown.

**Comment by pavankumarkatakam** [comment]
Thanks @vudung45 , Actually my data consists of more than 1000's of faces.

**Comment by vudung45** [comment]
@pavankumarkatakam : you can try the 2nd approach out. 
Check out my issue: https://github.com/davidsandberg/facenet/issues/324
I've scratched a way to find the best threshold distance for those 128D vectors. You can also check out dlib for python, they have a funciton that return 128D vectors with a threshold of 0.6 and the accuracy is claimed to be 99.38% (The model they use is probably somewhat similar or the same as inception resnet v1(v2)). 
Any input on the threshold and the pretrained model you are using would be much appreciated. I'm currently not able to download the data set to test this out.
Anyways, thank you and good luck!

**Comment by pavankumarkatakam** [comment]
Hi @vudung45, Once again thanks for your info. I have also used dlib for face detection previously in the scenario like a video frame/ image can consist a more than one faces but the facenet uses cafee models to extract only one face. But the dlib detection for partials faces is very poor. 

But if we use the dlib as the embedding creation then there is no matter of using the facenet. I have also seen the openface project but i chosen facenet because it is made up of tensorflow which can also run the GPU & TPU.

Please provide your email so that we can discuss more on this.

**Comment by pavankumarkatakam** [comment]
Hi @vudung45, did you mean `simple 2 layers neural network` is `fully connected layer`

**Comment by saisubramani** [comment]
hi guys i want to know that , how i can store the 128 embeddings in a nump file  for each person. i am beginner for this ... please help me
